saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/metadata/SAMLExtensionsParser.java

```diff
@@ -46,7 +45,7 @@ protected ExtensionsType instantiateElement(XMLEventReader xmlEventReader, Start
     }
 
     @Override
-    protected void processSubElement(XMLEventReader xmlEventReader, ExtensionsType target, JBossSAMLConstants element, StartElement elementDetail) throws ParsingException {
+    protected void processSubElement(XMLEventReader xmlEventReader, ExtensionsType target, SAMLMetadataQNames element, StartElement elementDetail) throws ParsingException {
         target.setElement(StaxParserUtil.getDOMElement(xmlEventReader));
```

Comment:
This needs revision, the `metadata` ExtensionsType should match that of `protocol`, and same for parsers